src/library.js

```diff
@@ -631,23 +615,15 @@ LibraryManager.library = {
     }
 #endif
 
-#if ASSERTIONS
-    var start = Date.now();
-#endif
-
-    if (!emscripten_realloc_buffer(newSize)) {
+    var replacement = emscripten_realloc_buffer(newSize);
+    if (!replacement) {
 #if ASSERTIONS
       err('Failed to grow the heap from ' + oldSize + ' bytes to ' + newSize + ' bytes, not enough memory!');
 #endif
       return false;
     }
 
-#if USE_PTHREADS
-    // Updating the views here is not strictly necessary, since we instrument each load and store
-    // to do so, but doing it here is clean and may be more efficient (avoid surprising the JIT
-    // later by taking a never-taken branch).
-#endif
-    updateGlobalBufferViews();
+    HEAPU32[DYNAMICTOP_PTR>>2] = requestedSize;
```

Comment:
why is this added? (it doesn't seem moved as I don't see a corresponding deletion?)

Reply:
Good point. The assignment occurs in sbrk() and brk(), and in our fork I have cleaned up those functions, but did not put the changes in this PR. Dropped this assignment, it's indeed unnecessary.